runbot_docker_orchest/models/runbot_repo.py

```diff
@@ -0,0 +1,69 @@
+from odoo import models, fields, api
+from dateutil import parser
+
+
+class RunbotRepo(models.Model):
+    _inherit = "runbot.repo"
+
+    is_quayio_image = fields.Boolean('Use quay.io image')
+    quayio_repo_name = fields.Char("Quay.io repo")
+
+    def create_build_orchest(self):
+        """Method create a build for image of orchestv
+        """
+        repo = self
+        settings = self.env['res.config.settings'].setting_quayio()
+        build_obj = self.env['runbot.build']
+        branch_obj = self.env['runbot.branch']
+        image_tags = settings.get_quayio_tags(
+            repo.quayio_repo_name, token=repo.token)
+        for tags in image_tags:
+            pull_image = False
+            branch = branch_obj.search([
+                ('name', '=', tags['name']), ('repo_id', '=', repo.id)])
+            if not branch:
+                branch = branch_obj.create(
+                    {'repo_id': repo.id, 'name': tags['name']})
+            build = build_obj.search([
+                ('branch_id', '=', branch.id),
+                ('name', '=', tags['manifest_digest'])])
+            if not build:
+                build_value = {
+                    'is_build_quayio': True,
+                    'branch_id': branch.id,
+                    'name': tags['manifest_digest'],
+                    'author': self.env.user.name,
+                    'author_email': self.env.user.email,
+                    'committer': settings.quayio_organization,
+                    'subject': tags['image_id'],
+                    'date': parser.parse(
+                        tags['last_modified']).strftime("%Y-%m-%d"),
+                    }
+                build_obj.create(build_value)
+                pull_image = True
+            if (repo.mode == 'hook' and branch.updatable):
+                pull_image = True
+                branch.sudo().updatable = False
+            if pull_image:
+                settings.pull_image(branch.pull_head_name)
+                pull_image = False
+
+    def _update_git(self):
+        if not self.is_quayio_image:
+            return super(RunbotRepo, self)._update_git()
+        self.create_build_orchest()
+
+    @api.onchange('quayio_repo_name', 'is_quayio_image')
+    @api.multi
+    def get_quayio_repo(self):
```

Comment:
Remove this onchange because this module will be generic for all `docker pull images` (not just quayio)